### PR TITLE
feat(ingestion/bigquery): Use sqlglot_lineage for usage and add more perf timers

### PR DESCRIPTION
**Pull request by shubhamjagtap639**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by asikowitz — CHANGES_REQUESTED** [review]
Can you break this down by both? i.e. one timer for sql parsing, another for store_usage_event?

**Review by asikowitz — APPROVED** [review]

File: metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py [hunk 1]
```diff
@@ -395,7 +398,10 @@ def resolve_table(self, table: _TableName) -> Tuple[str, Optional[SchemaInfo]]:
             if schema_info:
                 return urn_lower, schema_info
 
-        return urn_lower, None
+        if self.platform in CASE_SENSITIVE_PLATFORMS:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/utilities/sqlglot_lineage.py:401]: rename to TABLE_CASE_SENSITIVE_PLATFORMS

File: metadata-ingestion/src/datahub/utilities/bigquery_sql_parser.py [hunk 2]
```diff
@@ -1,92 +1,23 @@
...
-        Reason: in case table name contains hyphens which breaks sqllineage later on
-        Note: ignore cases of having keyword FROM as part of datetime function EXTRACT
-        """
-        return re.sub(
-            r"(?<!day\s)(?<!(date|time|hour|week|year)\s)(?<!month\s)(?<!(second|minute)\s)(?<!quarter\s)(?<!\.)(from\s)([^`\s();]+)",
-            r"\3`\4`",
-            sql_query,
-            flags=re.IGNORECASE,
-        )
+class BigQuerySQLParser:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/utilities/bigquery_sql_parser.py:7]: add a comment that this is unused and purely for tests for now

maybe move to the tests/ directory?

**Review by hsheth2 — APPROVED** [review]

**Comment by hsheth2** [comment]
@asikowitz would love to get this merged since the changes are good and I don't want them to get stale

**Comment by asikowitz** [comment]
Thanks for the reminder

**Pull request merged**